pandas/core/indexes/interval.py

```diff
@@ -819,7 +825,12 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
             return np.arange(len(self), dtype='intp')
 
         if self.is_non_overlapping_monotonic:
-            start, stop = self._find_non_overlapping_monotonic_bounds(target)
+            try:
+                start, stop = (
+                    self._find_non_overlapping_monotonic_bounds(target)
+                )
+            except TypeError:
+                return np.repeat(np.int(-1), len(target))
```

Comment:
@jschendel I am not fully sure this will cover all cases (but what was there before also not). 

`target` can in principle a mixture of valid and invalid keys. So maybe the easiest would be to fall back to the `else` branch that iterates through the elements separately in case this raises a TypeError.
That could be done by putting a `pass` here, and putting the `return` value of three lines below in the else part of a `try/except/else`.

Reply:
Sorry getting a bit confused here. You're saying that we should pass on line 833 here and left the code run to the else branch starting on line 851 (non IntervalIndex) where it loops over each element and appends -1 to the list if a KeyError is raised (I would probably add TypeError too).

Reply:
Yes, that is basically what I meant I think.

So that if you do `idx.get_indexer(['a', 1])` (where 1 is a valid key), you get [-0, 1] as result instead of [-1, -1] (assuming that `idx.get_loc(1)` would return 1)

Reply:
OK I've just pushed a commit implementing this. I had to make a few tweaks to the "non IntervalIndex" else branch (starting line 849 in core/indexes/interval.py) to make my two new tests pass, but hopefully haven't broken anything. See 2c48272